src/bun.js/webcore/streams.zig

```diff
@@ -1533,7 +1533,7 @@ pub const SinkDestructor = struct {
     const Subprocess = JSC.API.Bun.Subprocess;
     pub const Ptr = bun.TaggedPointerUnion(.{
         Detached,
-        Subprocess,
+        Subprocess.Island,
```

Comment:
Since this is now a separate pointer, we can avoid going through this code path and instead directly store it on the value. We don't need a tagged pointer union for this anymore.

Reply:
can i do this refactor in a followup?